<!--
-->
<launch>
  <arg name="robot_x" default="1.5"/>
  <arg name="robot_y" default="2.0"/>
  <arg name="robot_z" default="0.5"/>
  # Robot position. You better use a good odometry estimator as LOAM
  <node pkg="tf" type="static_transform_publisher" name="odom_tf" args="$(arg robot_x) $(arg robot_y) $(arg robot_z) 0 0 0 map base_link 10" />

  <arg name="target_x" default="1.25"/>
  <arg name="target_y" default="7.2"/>
  <arg name="target_z" default="3.0"/>

  # Flight 1 map conf
  <arg name="map" default="stage_1" />
  <!-- <arg name="map_path" default="$(find catenary_checker)/test/$(arg map)"/> -->
  <arg name="map_path" default="$(find marsupial_optimizer)/maps/$(arg map).bt"/>



  # Launch Octomap
  <arg name="resolution" default="0.05"/>
  <node name="octomap_server" type="octomap_server_node" pkg="octomap_server" args="$(arg map_path)">
    <param name="resolution" value="$(arg resolution)"/>
    <param name="frame_id" value="map"/>
  </node>

  <arg name="plane_dist" default="0.1"/>
  <arg name="dbscan_epsilon" default="0.1"/>
  <arg name="dbscan_gamma" default="0.1"/>
  <arg name="dbscan_theta" default="0.1"/>
  <arg name="use_dbscan_lines" default="true"/>
  <arg name="dbscan_min_points" default="5"/>
  <arg name="n_theta" default="6"/>
  <arg name="test_problems" default=""/>
  <arg name="use_parabola" default="true"/>
  <arg name="filename" default="$(find marsupial_optimizer)/maps/$(arg map).tar.gz"/>
  <arg name="debug" default="false"/>
  <arg name="pref_debug" default="" unless="$(arg debug)"/>
  <arg name="pref_debug" default="xterm -e gdb --args" if="$(arg debug)"/>
  <node name="test_ps" pkg="catenary_checker" type="test_preprocessed_scenario" output="screen" args="$(arg filename) $(arg test_problems)"  launch-prefix="$(arg pref_debug)" >
    <rosparam command="load" file="$(find marsupial_optimizer)/cfg/$(arg map)_ws.yaml"/>
    <remap from="point_cloud" to="octomap_point_cloud_centers"/>
    <param name="plane_dist" value="$(arg plane_dist)"/>
    <param name="dbscan_min_points" value="$(arg dbscan_min_points)"/>
    <param name="dbscan_epsilon" value="$(arg dbscan_epsilon)"/>
    <param name="dbscan_gamma" value="$(arg dbscan_gamma)"/>
    <param name="dbscan_theta" value="$(arg dbscan_theta)"/>
    <param name="use_dbscan_lines" value="$(arg use_dbscan_lines)"/>
    <remap from="points2" to="/octomap_point_cloud_centers"/>
    <param name="n_theta" value="$(arg n_theta)"/>
    <param name="plane_dist" value="$(arg plane_dist)"/>
    <param name="preprocessed_file" value=""/>

  </node>

  <node name="rviz" type="rviz" pkg="rviz" args="-d $(find catenary_checker)/viz/catenary_checker.rviz"/>

  <node pkg="catenary_checker" type="publish_pose.sh" name="pose_pub" args="$(arg target_x) $(arg target_y) $(arg target_z) map 5"/>
</launch>	

